### ISSUE TITLE
fix(stats): score multi-label annotations per label instead of as composite categories

### ISSUE THREAD

**Issue by tsenoner** [tsenoner/protspace_d3#419]
## Problem

Cluster-validity scoring treats a multi-valued annotation cell as one atomic category. `_category()` (`apps/protspace/src/protspace/stats/annotation_select.py:40`) strips the `|evidence` suffix per `;`-separated entry but **re-joins on `;`**, so:

```
Secreted|EXP;Nematocyst|IC   →   category "Secreted;Nematocyst"
```

This is Label Powerset by side effect, not by design. It is a valid partition, so the arithmetic is well-defined — but it fragments real categories into pseudo-categories that no legend row can match.

### Symptoms

- **Grey, un-hoverable dots on the legend score strips.** 90 of 544 per-category rows (16.5%) in the shipped `venom_eat_stats.parquetbundle` carry a category with no legend row, so `annotationCategoryScores` falls back to `#888` and strip↔legend hover sync does nothing for them.
- **Distorted scores.** In the demo, `Secreted` scores −0.630 and `Secreted;Nematocyst` −0.752 purely because 19 proteins were carved out of the blob they sit inside.
- **Silently narrowed `--stats-annotation auto`.** Composites inflate the distinct-value count against the `max_card` cap, so high-multiplicity columns can drop out of auto-selection.

## Fix: nearest-rival multi-label silhouette

A protein carrying `{A, C, D}` is a **member of all three**. When scoring its membership in A:

- `a(i)` = mean distance to the other members of A
- `b(i)` = **min** over categories the point does *not* carry, of the mean distance to that category

So C and D are excluded from its rival set — the point genuinely *is* a C and a D, and being near them is correct, not a separation failure.

**This is a strict extension, not a new metric.** For a single-label point, "categories I don't carry" is "all other categories", so it reduces to standard silhouette exactly (measured maxdiff `0.00e+00` across 12 real annotation × projection combinations). 12 of the 14 scored annotations in the demo bundle come out byte-identical.

It also scores `Secreted` correctly at −0.624 over all 811 proteins, runs at **0.5–0.6× current cost** (one distance matrix + one n×k matmul), and preserves the load-bearing `aggregate == size-weighted mean of parts` contract.

### Rejected alternatives, with the reason

| Option | Why not |
|---|---|
| **One-vs-rest** (`b` = mean over all non-A points) | Monotone in `b`, so **OvR ≥ multi-class always** — measured +0.34 (k=4) to +1.06 (k=28), flipping `family` from −0.41 to +0.65 on identical data. Does not preserve ranking (Spearman 0.20–0.66), which is the score strip's only job. Undefined for `Secreted` (811/811 coverage). |
| **Duplicate the row into each label** | Provably biased down: `a(i)` unchanged, `b(i)` deflated by exactly `m/(m+1)` by the zero-distance twin. A protein whose second label is unique scores exactly **−1.0**. |
| **First label only** | Silently wrong; manufactures categories from token order with no UI signal. |
| **Drop multi-label proteins** | Measured selection bias: multi-label proteins have 1.16–1.48× higher UniProt `annotation_score` and are 1.5–2.2× more likely to carry protein-level evidence (p to 9.6e-75). Confounds separation with curation depth. On the demo it collapses the annotation entirely (792/811 are pure `Secreted`). |
| **Skip multi-valued columns** | Removes `cc_subcellular_location` and `go_cc` from the demo — the two a venom biologist opens first — for a fix that would be reverted. |

## Calinski-Harabasz: suppress, do not extend

CH rests on the ANOVA identity `total = within + between`, which genuinely breaks under shared membership (measured 4280.9 vs 4548.8). Computing it on the mono-label subset would put three numbers side by side where two describe all proteins and one describes a biased subset. **Suppress CH whenever any protein in the annotation carries more than one label.**

## Scope line

A protein now counts once per label, so "N proteins scored" is ambiguous. Report separately, and only when they differ:

- single-label (unchanged): `42 categories · 811 proteins scored`
- multi-label: `42 categories · 811 proteins · 830 category assignments`

## Changes

All backend. **No frontend change required** — categories arrive atomic, so strip dots map 1:1 to legend rows and the aggregate row survives, keeping `hasAnnotationStats` true.

- `stats/annotation_select.py`
  - `_category()` → add `_categories()` returning `tuple[str, ...]`; keep `_category` for the ARI/NMI powerset path
  - `_clean()`, `_is_suitable_column()` — count **atomic** labels against the cardinality cap
  - `build_annotation_labels()` — returns `dict[str, dict[str, tuple[str, ...]]]` (the wire-level change)
  - `pair_by_id()` — leave on the powerset collapse and document it; ARI/NMI over a powerset partition is well-defined, just finer than the legend implies. Omega index is a separate, optional, later decision.
- `stats/base.py` — `StatContext.annotations` type follows
- `stats/metrics/annotation_validity.py`
  - `_per_category_silhouette()` — nearest-rival (`pairwise_distances_chunked` at 570k scale); guard the point carrying every category (empty rival set → NaN) as singletons are guarded today
  - `_per_category_davies_bouldin()` — extend the same way; DBI degrades monotonically and finitely under shared membership, and the existing `np.allclose(separation, 0)` short-circuit covers the degenerate limit
  - `compute()` — suppress CH per above
- `stats/metrics/validity.py` — `_emit_labeling` passes a plain `{id: str}`; adapt to the new shape

## Release / data impact

Touches `apps/protspace/**`, so merging cuts a PyPI release. Regenerating `apps/web/public/data/venom_eat_stats.parquetbundle` changes rows for exactly two annotations (`cc_subcellular_location`, `go_cc`); the other 12 are byte-identical.

## Verification

- Assert the reduction property directly: on every single-label annotation, nearest-rival output must equal `sklearn.silhouette_samples` exactly.
- Pin the `Secreted` case (811/811 coverage) — the one that breaks one-vs-rest.
- Pin that CH is absent for a multi-valued annotation and present for a single-label one.
- Extend `tests/contract/emit_bundles.py` with a multi-valued annotation so the seam is covered.

---

Found while reviewing #389. Backed by a multi-agent research pass whose adversarial stage overturned its own first recommendation (one-vs-rest) on measured data.
